cpp/splinepy/splines/bezier.hpp

```diff
@@ -345,6 +345,133 @@ class Bezier : public splinepy::splines::SplinepyBase,
         this->Base_::operator+(static_cast<Bezier<para_dim, dim>&>(*a)));
   }
 
+  template<bool is_rational, size_t inner_para_dim>
+  std::vector<std::shared_ptr<SplinepyBase>> ConvertComposeToBase(
+      const std::shared_ptr<SplinepyBase>& inner_function) const {
+    if constexpr (is_rational) {
+      auto super_return = this->ComposeSensitivity(
+          static_cast<RationalBezier<inner_para_dim, para_dim>&>(
+              *inner_function));
+
+      std::vector<std::shared_ptr<SplinepyBase>> to_return(super_return.size());
+
+      for (size_t i{}; i < super_return.size(); ++i) {
+        to_return[i] = std::make_shared<RationalBezier<inner_para_dim, 1>>(
+            super_return[i]);
+      }
+      return to_return;
+    } else {
+      auto super_return = this->ComposeSensitivity(
+          static_cast<Bezier<inner_para_dim, para_dim>&>(*inner_function));
+
+      std::vector<std::shared_ptr<SplinepyBase>> to_return(super_return.size());
+
+      for (size_t i{}; i < super_return.size(); ++i) {
+        to_return[i] =
+            std::make_shared<Bezier<inner_para_dim, 1>>(super_return[i]);
+      }
+      return to_return;
+    }
+  }
+
+  /// Spline composition.
+  /// inner_function requirements:
+  ///   1. Bezier Types
+  ///   2. dim is same as outer_function's par_dim
+  virtual std::vector<std::shared_ptr<SplinepyBase>>
+  SplinepyComposeSensitivities(
```

Comment:
Either both `ComposeSentitivieis` & `ComposeDerivative` or `Composition---`. I am leaning towards the second one